Website/index.js

```diff
@@ -0,0 +1,474 @@
+                flagDuty = "<p><span class='fw-bold'></span><span class='badge bg-warning mb-1'>Flag Duty</span><br><span class='fw-bold'>Duty: </span><span id=''>Singapore Flag</span></p>";
+            } else if (String($("#name").val()) == flagData[v]["school_flag"]) {
+                flagDuty = "<p><span class='fw-bold'></span><span class='badge bg-warning mb-1'>Flag Duty</span><br><span class='fw-bold'>Duty: </span><span id=''>School Flag</span></p>";
+            }
+
+        }
+        if (dd.getDay() === 6 || dd.getDay() === 0 || jsonData["dates"][v] === 11 || dd < first_date || dd > lastDate) {
+            var b = new Date(v);
+            var a = '<div id="' + per_dict[i]["id"] + '" class="container shadow pt-3 pb-3 mb-3 bg-white rounded-3"><h4>' + days[b.getDay()] + ", " + String(b.getDate()) + " " + months[b.getMonth()] + '</h4><p><span class="fw-bold"></span>No data is available for this day</span></p></div>';
+            $(a).appendTo("#personalised");
+        } else {
+            $("#weekend").hide();
+            var b = new Date(v);
+            var a = '<div id="' + per_dict[i]["id"] + '" class="container shadow pt-3 pb-3 mb-3 bg-white rounded-3"><h4>' + days[b.getDay()] + ", " + String(b.getDate()) + " " + months[b.getMonth()] + '</h4>' + flagDuty + '<p><span class="fw-bold"></span>' + recessBadge[nameData[String($("#name").val())][String(jsonData["dates"][v])]["recess"]] + '<br><span class="fw-bold">Duty: </span><span id="">' + duty_trans[nameData[String($("#name").val())][String(jsonData["dates"][v])]["duty"]] + '</span></p></div>';
+            var c = '<div id="' + per_dict[i]["id"] + '" class="container shadow pt-3 pb-3 mb-3 bg-white rounded-3"><h4>' + days[b.getDay()] + ", " + String(b.getDate()) + " " + months[b.getMonth()] + '</h4>' + flagDuty + '<p><span class="fw-bold"></span><span class="badge bg-warning mb-1">Morning Duty</span><br><span class="fw-bold">Duty: </span><span id="">' + duty_trans[nameData[String($("#name").val())]["morning_duty"]["duty"]] + '</span></p><p><span class="fw-bold"></span>' + recessBadge[nameData[String($("#name").val())][String(jsonData["dates"][v])]["recess"]] + '<br><span class="fw-bold">Duty: </span><span id="">' + duty_trans[nameData[String($("#name").val())][String(jsonData["dates"][v])]["duty"]] + '</span></p></div>';                    
+
+            if (nameData[String($("#name").val())]["morning_duty_check"]) {
+                $(c).appendTo("#personalised");
+            } else {
+                $(a).appendTo("#personalised");
+            }
+
+        }
+    }
+        counter += 1
+    })
+
+    localStorage.setItem("name", String($("#name").val()));
+    $("<div id='date2' class='text-center'><h4>Hello " + String($("#name").val()) + "!</h4><p>Here is your duty for the week you selected</p></div>").insertBefore("#date");
+    $("#personalised").show();
+        $(["#date", "#date2", "#day_1", "#day_2", "#day_3", "#day_4", "#day_5"]).each(function (i,v) {$(v).hide()});
```

Comment:
Codacy has a fix for the issue: [Missing semicolon.](https://app.codacy.com/gh/Davianlols6/SC-Duty-List/pullRequest?prid=8386818)
```suggestion
        $(["#date", "#date2", "#day_1", "#day_2", "#day_3", "#day_4", "#day_5"]).each(function (i,v) {$(v).hide();});
```